### DISCUSSION TITLE
Update v6.6

### DISCUSSION THREAD

**Discussion by TBog** [TBog/TBLauncher#187] [Release]
* fix horizontal widget page position
* cache icon pack icons to database for faster startup
* fix result list animations

<hr /><em>This discussion was created from the release <a href='https://github.com/TBog/TBLauncher/releases/tag/v6.6'>Update v6.6</a>.</em>

**Comment by heiwiper** [comment]
Since version 6.5 the keyboard isn't following the search bar on the first tap which require tapping twice for the keyboard to start following the search bar.
Here's a gif if I wasn't clear :
![tblauncher-bug](https://user-images.githubusercontent.com/12894695/130629831-745a17f5-fa59-4f20-af63-bdedc1503c3f.gif)

**Reply by TBog** [reply]
Thanks for noticing this issue. I'll look into it.

**Reply by TBog** [reply]
The issue appeared with commit 76423d7cb37b8d1e962cd5773a49b9b8865745ee
I can undo this commit, but I still need to find a way to not update the result adapter twice when the launcher resumes.

**Reply by heiwiper** [reply]
Did you undo that commit in the last pre-release ?
The problem is still persistent, I don't know much about Android development but what I understand is that the the keyboard should show up right after the launcher resumes which is when the user leaves an app. in this case it's working just fine on the last pre-release, what I pointed to is when the initial desktop is the widget page and that the user taps once to switch to the search page, this still doesn't show up the keyboard on the first tap.
I would also like to point out that I updated from version 6.4 to 6.5 directly and I didn't try the pre-release versions so I can't say exactly in which release version it got broken, all I know is it was working on the sable release 6.4.

**Reply by ltguillaume** [reply]
@HeiWiper I can't reproduce this, but I am using a custom keyboard app. Perhaps this has something to do with it, otherwise it could be some combination of settings (I just found an issue there as well).

**Reply by TBog** [reply]
v6.6-dev.1 has a fix keyboard not showing up after onResume

I don't seem to have the problem you describe. Until I can reproduce it I can't really do anything except ask you to test my changes.

**Reply by heiwiper** [reply]
I use a custom keyboard but I also tried with AOSP keyboard and had the same result.
I will backup my settings and try to reproduce it on a fresh install.

**Reply by TBog** [reply]
Send the backup my way also if you can.

**Reply by heiwiper** [reply]
# Steps
- install app
- Go to **TinyBit settings** > **Behaviour**
- Under **Behaviour** section click on **Home Button** and check **Show "Widget" desktop**
- Press home button, it should take you to the widget page
- Tap to switch to search page and that's where the keyboard won't show up
- Switching to widget page and back to search page should show the keyboard

let me know if the backup is still required

**Reply by ltguillaume** [reply]
Still cannot reproduce, sorry.

**Reply by TBog** [reply]
I did reproduce it. I'm working on it now.

**Comment by ltguillaume** [comment]
Maybe it would be better to put the `Links` block on the app context menu at the top? The menu is pretty long, so on somewhat smaller screens (when the keyboard's out) you'll need to scroll to get to the more often used items (such as App info and - for me at least - Hibernate).

**Reply by TBog** [reply]
Ok, ok, I'll see how to make the groups order in that menu customizable.
This is not a high priority so I'll need to add an issue to remember it.

**Reply by ltguillaume** [reply]
I don't know if it needs to be customizable, I just thought it could be in the order of how likely ppl use their items often (in which case I'd even say Links - Customize - Preferences, so even more shuffling, lolz).

**Reply by TBog** [reply]
I'm sure you feel that order is the best, but I don't. That's why I wanted to add the order preference.
In the mean time you can fork the repo and change the order.

**Comment by ltguillaume** [comment]
I just found that the combination `Link keyboard to search bar visibility ENABLED` + `Close keyboard same as back button DISABLED` + `Desktop mode Search -> Back button = Show Widget desktop` now does not go to the Widget desktop anymore when pressing back twice. It just stays on the Search desktop with the keyboard closed.

**Reply by TBog** [reply]
Both on the emulator and my device I can't close the keyboard with the settings you set.
I found a possible way to show the "Search" desktop with the keyboard closed (using the bug found by @HeiWiper ) and indeed the back button does nothing in that case.
1. I fixed the bug found by @HeiWiper and so the keyboard should not close when `Link keyboard to search bar visibility ENABLED` + `Close keyboard same as back button DISABLED`
2. In care you have the keyboard closed and you press "back" then execute the back button action.
